Bio/GenBank/Scanner.py

```diff
@@ -1772,9 +1772,6 @@ def _feed_header_lines(self, consumer, lines):
                                     ),
                                     data,
                                 )
-                                structured_comment_dict.setdefault(
-                                    structured_comment_key, OrderedDict()
-                                )
```

Comment:
That's a subtle change, I wonder if it makes any difference to the final annotation on the SeqRecord object... and if anyone would care?

Reply:
Not sure I understand exactly what you are saying here. Should the setdefault remain for some reason?
`structured_comment_dict` is a defaultdict at this point.

Reply:
I don't actually know what the public object is in the end.

Reply:
It gets passed to the consumer via the ``.structured_comment`` method, defined here:

https://github.com/biopython/biopython/blob/biopython-179/Bio/GenBank/__init__.py#L1062

and:

https://github.com/biopython/biopython/blob/biopython-179/Bio/GenBank/__init__.py#L1587

So the (Ordered)Dict was in the returned annotation. I'd be happier to replace it with a plain dict, we don't know what people might be doing with it (e.g. modifying the dict or pulling potentially missing data out of it).

Reply:
I mean it ought to be easy to change from the old behaviour giving an OrderedDict of OrderedDicts, to returning a dict of dicts (not using defaultdict at all).

Reply:
Looking at https://docs.python.org/3/library/stdtypes.html#dict.setdefault I was probably worrying about nothing, the old behaviour had the default set, so does the PR.